cmd/cigarToBed/cigarToBed.go

```diff
@@ -0,0 +1,198 @@
+			ChromEnd = ChromStart + int(aln[i+1].RunLength)                                                             //get the last position that is I
+			ins := bed.Bed{Chrom: Chrom, ChromStart: ChromStart, ChromEnd: ChromEnd, Name: "ins", FieldsInitialized: 4} //TODO: now just write "chr1", but find a way to grab from fasta header?
+			bed.WriteBed(insBed.File, ins)
+		}
+		if aln[i].Op != 2 { //in insertion bed, only need to update ChromCurrent if the cigar fragment is M or I, not D
+			ChromCurrent += int(aln[i].RunLength)
+		}
+	}
+	//deletion bed
+	delBed := fileio.EasyCreate(outDel_bed) //TODO: add flag for output file name
+	defer delBed.Close()
+	//TODO: now have to write FirstPos, but find a way to grab from fasta header
+	ChromCurrent = FirstPos_DelBed - 1 //reset the same 3 variables for position
+	ChromStart = FirstPos_DelBed - 1
+	ChromEnd = FirstPos_DelBed - 1
+	for i := 0; i < len(aln)-1; i++ {
+		if aln[i].Op == 0 && aln[i+1].Op == 1 {
+			ChromStart = ChromCurrent + int(aln[i].RunLength) //the position before I starts
+			ChromEnd = ChromStart + 1                         //add 1 so that the length of the bed entry is 1
+			del := bed.Bed{Chrom: Chrom, ChromStart: ChromStart, ChromEnd: ChromEnd, Name: "del", FieldsInitialized: 4}
+			bed.WriteBed(delBed.File, del)
+		}
+		if aln[i].Op != 1 { //in deletion bed, only update ChromCurrent if the cigar fragment is M or D, not I
+			ChromCurrent += int(aln[i].RunLength)
+		}
+	}
+
+	//visualize
+	visualize := align.View(faOne.Seq, faTwo.Seq, aln)
+	fmt.Println(visualize)
+
+	//raven added this block to put visualized alignment output into MSA Fasta
+	if outFa != "" { //only write to file if a filename is given in the faOut command line option
+		outFaFile, err := os.Create(outFa) //fileio.EasyCreate and other fileio tools should work, but I don't really know how to use them
+		if err != nil {
+			log.Fatalf("Write to file failed on step 1\n")
+		}
+		visualizeOutput := ">" + faOne.Name + "\n" + strings.Split(visualize, "\n")[0] + "\n" + ">" + faTwo.Name + "\n" + strings.Split(visualize, "\n")[1] + "\n"
+		_, err = outFaFile.WriteString(visualizeOutput)
+		if err != nil {
+			log.Fatalf("Write to file failed on step 2\n")
+		}
+		outFaFile.Close() //commented out defer close
+	}
+
+	//cigar to graph
+	//genomeGraph := cigarToGraph(faOne, faTwo, aln)
+	//genomeGraph.PrintGraph(genomeGraph)
+}
+
+//raven edited this block to specify only 1 sequnce is expected in each fasta file and add Usage nad options
+func usage() {
+	fmt.Print(
+		"./cigarToBed\n" +
+			" Uses globalAlignment, affineGap (instead of constGap) to align 2 .fasta files, each with only 1 sequence, then convert cigars to ins and del beds\n" +
+			"Usage:\n" +
+			" cigartobed target.fasta query.fasta\n" +
+			"options:\n" +
+			"	-faOut=fasta MSA output filename\n" +
```

Comment:
I actually don't think you'll need to write out all these flags (lines 159-164). I believe that the flag.String and flag.Int that you use on lines 173-177 will actually trigger the info on those lines to print out with the usage anyways. I was just looking at this and testing how the usage of other programs print out and that's how I figured this out. I too wrote out lines like you have here and then deleted them when I saw how the flag functions end up all coming together. But someone more versed in this should weigh in and make sure I'm correct.

Reply:
You are right. The call to `flag.PrintDefaults()` on line 165 handles this. If you do not like the defaults for whatever reason you could make your own like you have now, but doing both is unnecessary.

Reply:
Good to know! Just fixed this.